setup.py

```diff
@@ -315,7 +315,7 @@ def run(self):
     'pandas',
 ]
 
-NUMPY_STR = 'numpy >= 1.17.0'
+NUMPY_STR = 'oldest-supported-numpy'
```

Comment:
```suggestion
#
# The numpy version here is especially significant because it affects our ability to build
# wheels that are backwards-compatible with older numpy versions. If restrict this to
# e.g. numpy>=1.17.0, then our wheel building process becomes significantly more
# complicated. Related issue: #3266.
# 
NUMPY_STR = 'oldest-supported-numpy'
```